app/src/validators/common.js

```diff
@@ -45,15 +45,27 @@ const type = {
     version: 'uuidv4'
   }),
 
-  metadata: (minKeyCount = 0, minValueStringLength = 0) => Joi.object()
+  /**
+   * custom metadata (object) type schema with parameters
+   * @param minKeyCount minimum number of tags allowed
+   * @param minValueStringLength minimum string length of tag value allowed
+   * @param maxKeyCount maximum number of tags allowed (default ot 9 because COMS also adds a `coms-id` tag by default)
+   */
+  metadata: ({ minKeyCount = 0, minValueStringLength = 1 } = {}) => Joi.object()
     .pattern(/^x-amz-meta-.{1,255}$/i, Joi.string().min(minValueStringLength).max(255), { matches: Joi.array().min(minKeyCount) })
     .unknown(),
 
-  tagset: (minKeyCount = 1, minValueStringLength = 0) => Joi.object()
+  /**
+   * custom tagset (object) type schema with parameters
+   * @param minKeyCount minimum number of tags allowed
+   * @param minValueStringLength minimum string length of tag value allowed
+   * @param maxKeyCount maximum number of tags allowed (default ot 9 because COMS also adds a `coms-id` tag by default)
```

Comment:
JSDocs do not match function signature. Params should be marked as optional (brackets) with a default value specified. Should also document `@returns object Joi object` or similar.